### PR TITLE
nixos/rtkit: Add option for rtkit-daemon command-line args

### PR DESCRIPTION
**Pull request by rvl**
## Description of changes

When configuring rtkit for realtime audio with pipewire (following [FAQ: How to change rtkit rtprio](https://gitlab.freedesktop.org/pipewire/pipewire/-/wikis/FAQ#how-to-change-rtkit-rtprio)) , I found it necessary to adjust the command-line arguments of `rtkit-daemon`.

This option makes it easier to do.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: nixos/modules/security/rtkit.nix [hunk 1]
```diff
@@ -20,19 +24,38 @@ with lib;
...
+
+    services.dbus.packages = [ package ];
 
-    systemd.packages = [ pkgs.rtkit ];
+    systemd.packages = [ package ];
 
-    services.dbus.packages = [ pkgs.rtkit ];
+    systemd.services.rtkit-daemon = {
+      serviceConfig.ExecStart = [
+        ""
```
Comment (FliegendeWurst) [inline nixos/modules/security/rtkit.nix:55]: Why is there an empty string here?

Comment (FliegendeWurst) [reply nixos/modules/security/rtkit.nix:55]: Nevermind, I figured it out. It is to override the default.

File: nixos/modules/security/rtkit.nix [hunk 2]
```diff
@@ -20,19 +24,38 @@ with lib;
...
+    environment.systemPackages = [ package ];
+
+    services.dbus.packages = [ package ];
 
-    systemd.packages = [ pkgs.rtkit ];
+    systemd.packages = [ package ];
 
-    services.dbus.packages = [ pkgs.rtkit ];
+    systemd.services.rtkit-daemon = {
+      serviceConfig.ExecStart = [
```
Comment (FliegendeWurst) [inline nixos/modules/security/rtkit.nix:54]:
```suggestion
      # Note: the first "" overrides the ExecStart from the upstream unit
      serviceConfig.ExecStart = [
```

Comment (rvl) [reply nixos/modules/security/rtkit.nix:54]: I have added a comment on the `""`.

**Review by FliegendeWurst — APPROVED** [review]

**Comment by FliegendeWurst** [comment]
@ofborg build nixosTests.rtkit

**Comment by rvl** [comment]
Thanks @FliegendeWurst and @wegank.

**Pull request merged**